Sources/Prometheus/PrometheusCollectorRegistry.swift

```diff
@@ -93,6 +93,18 @@ public final class PrometheusCollectorRegistry: Sendable {
         }
     }
 
+    /// Creates a new ``Counter`` collector or returns the already existing one with the same name,
+    /// based on the provided descriptor.
+    ///
+    /// When the ``PrometheusCollectorRegistry/emit(into:)`` is called, metrics from the
+    /// created ``Counter`` will be part of the export.
+    ///
+    /// - Parameter descriptor: An ``FQMetricDescriptor`` that provides the fully qualified name for the metric.
+    /// - Returns: A ``Counter`` that is registered with this ``PrometheusCollectorRegistry``
+    public func makeCounter(descriptor: FQMetricDescriptor) -> Counter {
+        return self.makeCounter(name: descriptor.name)
```

Comment:
In a follow up PR will adjust this to enable printing the `HELP` line if applicable for all cases.